File: 2079. Watering Plants/Solution.py
from typing import List


class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        res, water = len(plants), capacity
        for i, need in enumerate(plants):
            if water < need:
                res += i * 2
                water = capacity
            water -= need
        return res

File: 2079. Watering Plants/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity)
    {
        int res = plants.size(), water = capacity;
        for (size_t i = 0; i < plants.size(); i++) {
            if (water < plants[i]) {
                res += 2 * i;
                water = capacity;
            }
            water -= plants[i];
        }
        return res;
    }
};
